Dylan-Clarry/github-issue-templates
# buggy

name: Pull Request
description: ""
title: Pull Request
labels: []
body:
  - type: textarea
    attributes:
      label: Description
      description: "Please include a summary of the change and which issue is fixed. Please also include relevent motivation and context. List any dependencies that are required for this change.

        Please delete options options that are not relevent and please make sure that you X the appropriate boxes.

        Fixes # (issue)"
      placeholder: ""
    validations:
      required: true
  - type: checkboxes
    attributes:
      label: Type of Change
      description: ""
      options:
        - label: Bug fix (non-breaking change which fixes an issue)
        - label: New feature (non-breaking change which adds functionality)
        - label: Breaking change (fix or feature that would cause existing functionality to not work as expected)
        - label: This change requires a documentation update
    validations:
      required: true
  - type: checkboxes
    attributes:
      label: Checklist
      options:
        - label: My code follows the style guidelines of this project
        - label: I have performed a self-review of my own code
        - label: I have commented my code, particularly in hard-to-understand areas
        - label: I have made corresponding changes to the documentation
        - label: My changes generate no new warnings
        - label: I have added tests that prove my fix is effective or that my feature works
        - label: New and existing unit tests pass locally with my changes
        - label: Any dependent changes have been merged and published in downstream modules
        - label: I have checked my code and corrected any misspellings
    validations:
      required: true
